src/components/Admin/AdminDataViews/AdminDataViews.tsx

```diff
@@ -0,0 +1,386 @@
+          addColumnDefinition({
+            variables: {
+              columnName: `${tableName}_column_${nanoid(8)}`,
+              tableName,
+            },
+          })
+        }}
+        isAdding={isAddingColumn}
+      />
+    </>
+  )
+}
+
+interface DataViewDisplayProps {
+  title: string
+  placeholder: string
+  loading: boolean
+  dropdownValue: string
+  options: DropdownItemProps[]
+  onChange: (_: unknown, value: DropdownProps) => void
+  data: object | undefined
+  dataName: string
+  onSave: (data: object) => void
+  isSaving: boolean
+  onDelete: () => void
+  isDeleting: boolean
+  onAdd: () => void
+  isAdding: boolean
+}
+
+const DataViewDisplay: React.FC<DataViewDisplayProps> = ({
+  title,
+  placeholder,
+  loading,
+  dropdownValue,
+  options,
+  onChange,
+  data,
+  dataName,
+  onSave,
+  isSaving,
+  onDelete,
+  isDeleting,
+  onAdd,
+  isAdding,
+}) => {
+  const { t } = useLanguageProvider()
+  const { ConfirmModal } = useConfirmationModal({
+    type: 'warning',
+    confirmText: t('BUTTON_CONFIRM'),
+  })
+
+  return (
+    <div>
+      <Header as="h2">{title}</Header>
+      <ConfirmModal />
+      <div className="flex-row-space-between">
+        <Dropdown
+          selection
+          clearable
+          placeholder={placeholder}
+          loading={loading}
+          value={dropdownValue}
+          options={options}
+          onChange={onChange}
+          style={{ minWidth: 300 }}
+        />
+        <div>
+          <Button
+            primary
+            inverted
+            disabled={!data}
+            loading={isDeleting}
+            icon={<Icon name="trash alternate outline" size="small" />}
+            content={t('DATA_VIEW_CONFIG_DELETE_BUTTON')}
+            onClick={onDelete}
+          />
+          <Button
+            primary
+            inverted
+            loading={isAdding}
+            icon={<Icon name="plus" size="tiny" color="blue" />}
+            content={t('DATA_VIEW_CONFIG_ADD_BUTTON')}
+            onClick={onAdd}
+          />
+        </div>
+      </div>
+      {data && (
+        <JsonEditor
+          data={data}
+          onSave={onSave}
+          isSaving={isSaving}
+          name={dataName}
+          collapsed={1}
+          displayArrayKey={false}
+          quotesOnKeys={false}
+          displayDataTypes={true}
+          style={{ padding: '10px' }}
+        />
+      )}
+    </div>
+  )
+}
+
+const getDataTableOptions = (
+  data: GetDataTablesQuery | undefined,
+  includeLookupTables: boolean,
+  t: TranslateMethod
+) => {
+  if (!data) return []
+
+  const options = (data.dataTables?.nodes as DataTable[])
+    .filter(({ isLookupTable }) => (includeLookupTables ? true : !isLookupTable))
+    .map(({ id, tableName, isLookupTable }) => {
+      const table = toCamelCase(tableName)
+      return {
+        key: `${table}_${id}`,
+        text: `${table}${isLookupTable ? ` (${t('LOOKUP_TABLE_TITLE')})` : ''}`,
+        value: table,
+      }
+    })
```

Comment:
We're relying on the "data_tables" being catalogued in the `data_table` table. This should happen automatically if the data table was created as a result of an action (modifyRecord) or imported as a lookup table, but might be missing if it was manually created. 

I also noticed some in there that shouldn't be (such as `userOrganisation`), so we should remove that from snapshots and check to see if it comes back.